assets/js/components/surveys/CurrentSurvey/index.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * CurrentSurvey component.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useMount } from 'react-use';
+import { Slide } from '@material-ui/core';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect, useDispatch } from 'googlesitekit-data';
+import { CORE_FORMS } from '../../../googlesitekit/datastore/forms/constants';
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+import SurveyCompletion from '../SurveyCompletion';
+import SurveyTerms from '../SurveyTerms';
+import SurveyQuestion from './SurveyQuestion';
+import {
+	getCurrentQuestionAndOrdinal,
+	getTriggeredCompletion,
+	isLastQuestion,
+} from './utils';
+import {
+	TYPE_MULTI_SELECT,
+	TYPE_OPEN_TEXT,
+	TYPE_RATING,
+	TYPE_SINGLE_SELECT,
+} from './constants';
+
+const SURVEY_ANSWER_DELAY_MS = 300;
+
+const KNOWN_QUESTION_TYPES = [
+	TYPE_MULTI_SELECT,
+	TYPE_OPEN_TEXT,
+	TYPE_RATING,
+	TYPE_SINGLE_SELECT,
+];
```

Comment:
Suggestion: move this to constants as `TYPES` or `QUESTION_TYPES` as an object. We could even export it with `Object.freeze` if we really want it to be immutable-ish.

We could still define this here if we wanted to with `Object.keys` or do that inline since I think we only use it once anyways?

Reply:
I've replaced this with a runtime check of `Object.values()`, this seems OK to me.

Reply:
Ah yes, that's what I meant 👍